basic_autonomy/src/basic_autonomy.cpp

```diff
@@ -397,7 +400,10 @@ namespace basic_autonomy
                 centerline_points.push_back(current_position);
             }
 
-            //Add points from following lanelet to provide sufficient distance for adding buffer
+            // Add points from the remaining length of the target lanelet to provide sufficient distance for adding buffer
+            centerline_points.insert(centerline_points.end(), downsampled_target_centerline.begin() + end_index_target_centerline, downsampled_target_centerline.end());
+            
+            // Also add points from the lanelet following the target lanelet to provide sufficient distance for adding buffer
             auto following_lanelets = wm->getMapRoutingGraph()->following(ending_lanelet, false);
```

Comment:
Seems like this additional buffer would not be needed if we already have the required length from line 404.
I guess its not entirely required to have this check, since the buffer would be removed anyway but we can probably avoid some cost with looking up a lanelet and extracting points in cases we dont need them